src/clover/simulation/storage.py

```diff
@@ -103,33 +107,139 @@ def battery_iteration_step(
             "Battery undefined despite an itteration step being called.",
         )
 
-    battery_energy_flow = float(battery_storage_profile.iloc[time_index, 0])  # type: ignore [arg-type]
-    if time_index == 0:
-        new_hourly_battery_storage = initial_battery_storage + battery_energy_flow
-    else:
+    def _charge_or_discharge(battery_energy_flow: float) -> tuple[float, float]:
+        """
+        Charge or discharge the battery by some amount.
+
+        :param: battery_energy_flow:
+            The energy flow into or out of the battery.
+
+        :returns:
+            - The new hourly battery storage;
+            - Any power which wasn't stored in the batteries or couldn't be discharged.
+
+        """
         # Battery charging
         if battery_energy_flow >= 0.0:
             new_hourly_battery_storage = float(
                 hourly_battery_storage[time_index - 1]
-            ) * (1.0 - minigrid.battery.leakage) + minigrid.battery.conversion_in * min(
-                battery_energy_flow,
-                minigrid.battery.charge_rate
-                * (maximum_battery_storage - minimum_battery_storage),
+            ) * (1.0 - minigrid.battery.leakage) + minigrid.battery.conversion_in * (
+                stored_power := min(
+                    battery_energy_flow,
+                    minigrid.battery.charge_rate
+                    * (maximum_battery_storage - minimum_battery_storage),
+                )
             )
+            un_accouted_power = battery_energy_flow - stored_power
         # Battery discharging
         else:
             new_hourly_battery_storage = hourly_battery_storage[time_index - 1] * (
                 1.0 - minigrid.battery.leakage
-            ) + (1.0 / minigrid.battery.conversion_out) * max(
-                battery_energy_flow,
-                (-1.0)
-                * minigrid.battery.discharge_rate
-                * (maximum_battery_storage - minimum_battery_storage),
+            ) + (1.0 / minigrid.battery.conversion_out) * (
+                discharged_power := max(
+                    battery_energy_flow,
+                    (-1.0)
+                    * minigrid.battery.discharge_rate
+                    * (maximum_battery_storage - minimum_battery_storage),
+                )
             )
+            un_accouted_power = battery_energy_flow + abs(discharged_power)
```

Comment:
Please change this variable name to something clearer

Reply:
Renamed this :smiley: